test/include/seqan3/test/performance/seqan2_minimiser.h

```diff
@@ -0,0 +1,221 @@
+#include <seqan3/core/platform.hpp>
+
+#ifdef SEQAN3_HAS_SEQAN2
+#include <seqan/seq_io.h>
+
+//!\brief Strong type for passing the window size.
+struct window { uint64_t v; };
+//!\brief Strong type for passing the kmer size.
+struct kmer { uint64_t v; };
+//!\brief Strong type for passing number of bins.
+struct bins { uint64_t v; };
+//!\brief Strong type for passing number of bits.
+struct bits { uint64_t v; };
+//!\brief Strong type for passing number of hash functions.
+struct hashes { uint64_t v; };
+
+//!\brief Whether to use xor for computing the hash value.
+enum use_xor : bool
+{
+    //!\brief Do not use xor.
+    no,
+    //!\brief Use xor.
+    yes
+};
+
+template<typename shape_t, use_xor do_xor = use_xor::yes>
+class minimiser
+{
+private:
+    //!\brief The alphabet type.
+    using alphabet_t = seqan::Dna;
+    //!\brief The text type.
+    using text_t = seqan::String<alphabet_t>;
+    //!\brief The type of the complemented text.
+    using complement_t = seqan::ModifiedString<text_t const, seqan::ModComplementDna>;
+    //!\brief The type of the reverse complemented text.
+    using reverse_complement_t = seqan::ModifiedString<complement_t, seqan::ModReverse>;
+
+    //!\brief The window size of the minimiser.
+    uint64_t w{};
+    //!\brief The size of the k-mers.
+    uint64_t k{};
+    //!\brief Random but fixed value to xor k-mers with. Counteracts consecutive minimisers.
+    uint64_t seed{};
+
+    //!\brief Shape for computing the forward strand k-mers.
+    shape_t forward_shape{};
+    //!\brief Shape for computing the reverse strand k-mers.
+    shape_t reverse_shape{};
+
+    //!\brief Stores the k-mer hashes of the forward strand.
+    std::vector<uint64_t> forward_hashes;
+    //!\brief Stores the k-mer hashes of the reverse complement strand.
+    std::vector<uint64_t> reverse_hashes;
+
+
+public:
+
+    //!\brief Stores the hashes of the minimisers.
+    std::vector<uint64_t> minimiser_hash;
+    //!\brief Stores the begin positions of the minimisers.
+    std::vector<uint64_t> minimiser_begin;
+    //!\brief Stores the end positions of the minimisers.
+    std::vector<uint64_t> minimiser_end;
+
+    minimiser() = default;                                        //!< Defaulted
+    minimiser(minimiser const &) = default;                       //!< Defaulted
+    minimiser(minimiser &&) = default;                            //!< Defaulted
+    minimiser & operator=(minimiser const &) = default;           //!< Defaulted
+    minimiser & operator=(minimiser &&) = default;                //!< Defaulted
+    ~minimiser() = default;                                       //!< Defaulted
+
+    /*!\brief Constructs a minimiser from given k-mer, window size and a seed.
+     * \param[in] w_    The window size.
+     * \param[in] k_    The k-mer size.
+     * \param[in] seed_ The seed to use. Default: 0x8F3F73B5CF1C9ADE.
+     */
+    minimiser(window const w_, kmer const k_, uint64_t const seed_ = 0x8F3F73B5CF1C9ADE) :
```

Comment:
You don't use this constructor anywhere? Either remove the constructor or use it and remove the resize function.